core/src/main/java/io/undertow/server/HttpServerExchange.java

```diff
@@ -336,6 +336,9 @@ public final class HttpServerExchange extends AbstractAttachable {
      */
     private InetSocketAddress destinationAddress;
 
+    private static final AtomicLong REQUEST_ID_GENERATOR = new AtomicLong(0);
```

Comment:
Is it OK from a security perspective to have predictable ids here? (It's only the pattern that is suspicious, not a fact I know for the case).

Reply:
That's a very good question! From the top of my head, now, it seems it is okay, but I'll do some research before considering this the final implementation for connection ids.